### PR TITLE
Feature/save poster function

### PR DESCRIPTION
**Pull request by DrewBradley**
Created savePoster function that pushes poster object to savedPosters array. How do we prevent pushing duplicate objects to the array.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -198,6 +198,23 @@ function userCreatedPoster() {
   showMainPoster();
 }
 
+function savePoster() {
+  var save = {
+    image: posterImage.src,
+    title: posterTitle.innerText,
+    quote: posterQuote.innerText,
+  }
+  for (var i = 0; i < savedPosters.length; i++){
```
Comment (GreysonElkins) [inline src/main.js:207]: Why are you using a for loop here?

When I run it, the object never makes it into the array. (How many times will this loop run when the array is empty?)

File: src/main.js [hunk 2]
```diff
@@ -198,6 +198,23 @@ function userCreatedPoster() {
...
+    if (savedPosters.includes(save.image && save.title && save.quote) === false) {
```
Comment (GreysonElkins) [inline src/main.js:208]: Let's say `save.img = 'turtle'`, `save.title = 'turtle'` and `save.title = 'Your code rocks!'`; wouldn't `savedPosters.includes` be checking that doesn't look like `['turtle', 'turtle', 'Your code rocks!']`?

**Review by GreysonElkins — COMMENTED** [review]
Y'all this codebase is looking _tite_! The functions are super bite-sized and the logic is surgical. I really dig how `createRandomPoster` is reassigning node values in the DOM. 

If you're looking for a challenge, would it be possible to combine `showPosterForm`, `showSavedPosters`, and `showMainPoster` into a single function using arguments? 

I was surprised to learn that you don't need a parameter to pass an event as an argument (as in `userCreatedPoster`) - very cool! I want to take the opportunity to point out that this could get sticky later if you wanted to use other arguments in addition to an event. 
```
function userCreatedPoster(dog = 'this should print') {
  event.preventDefault();
  console.log(dog)
```
I added the parameter `dog` and assigned a default value of `this should print`, and then logged the parameter. There's a problem when the function is triggered by `button.addEventListener('click', userCreatedPoster)` though - try it out and see what happens! 

image

Keep up the good work!

File: src/main.js (hunk 1 above)
Comment (gaj23) [reply src/main.js:207]: Thank you for the leading question! The example below helps me visualize things more, too. Much appreciated! ✨

**Review by gaj23 — APPROVED** [review]
I'm glad to have feedback from your rock, Drew! It's great to have someone outside of the project offer us a view from their POV. Based on his feedback, I think we could benefit from spending some time paring this function back to an if statement. 

Additionally, my mentor sent some great feedback via slack for refactoring. I would like to spend some time applying her suggestions, too. 

Great working with you last night. It felt productive and fun!

File: src/main.js (hunk 2 above)
Comment (DrewBradley) [reply src/main.js:208]: Thanks Greyson! You rock, rock! Thanks for pointing us in the right direction.

**Pull request merged**